src/content/contribute/writing-a-plugin.md

```diff
@@ -133,76 +148,63 @@ Once we can latch onto the webpack compiler and each individual compilations, th
 
-- __waterfall__ Plugins applied using
-
-`applyPluginsWaterfall(name: string, init: any, args: any...)`
-
-Here each of the plugins are called one after the other with the args from the return value of the previous plugin. The plugin must take the order of its execution into account.
-It must accept arguments from the previous plugin that was executed. The value for the first plugin is `init`. This pattern is used in the Tapable instances which are related to the `webpack` templates like `ModuleTemplate`, `ChunkTemplate` etc.
-
-- __asynchronous__ When all the plugins are applied asynchronously using
-
-`applyPluginsAsync(name: string, args: any..., callback: (err?: Error) -> void)`
-
-The plugin handler functions are called with all args and a callback function with the signature `(err?: Error) -> void`. The handler functions are called in order of registration. `callback` is called after all the handlers are called.
-This is also a commonly used pattern for events like `"emit"`, `"run"`.
-
-- __async waterfall__ The plugins will be applied asynchronously in the waterfall manner.
-
-`applyPluginsAsyncWaterfall(name: string, init: any, callback: (err: Error, result: any) -> void)`
-
-The plugin handler functions are called with the current value and a callback function with the signature `(err: Error, nextValue: any) -> void.` When called `nextValue` is the current value for the next handler. The current value for the first handler is `init`. After all handlers are applied, callback is called with the last value. If any handler passes a value for `err`, the callback is called with this error and no more handlers are called.
-This plugin pattern is expected for events like `"before-resolve"` and `"after-resolve"`.
-
-- __async series__ It is the same as asynchronous but if any of the plugins registered fails, then no more plugins are called.
-
-`applyPluginsAsyncSeries(name: string, args: any..., callback: (err: Error, result: any) -> void)`
-
--__parallel__ -
-
-`applyPluginsParallel(name: string, args: any..., callback: (err?: Error) -> void)`
+```javascript
+import { SyncHook, AsyncSeriesHook } from 'tapable';
+
+class SomeWebpackInternalClass {
+  constructor() {
+    this.hooks = {
+      // Create hooks:
+      compilation: new SyncHook(),
+      run: new AsyncSeriesHook(),
+    };
+  }
+  
+  someMethod() {
+    // Call a hook:
+    this.hooks.run.call();
+  
+    // Call another hook:
+    // (This is an async one, so webpack passes a callback into it)
+    this.hooks.run.callAsync(() => {
+      // The callback is called when all tapped functions finish executing
+    });
+  }
+}
+```
 
-`applyPluginsParallelBailResult(name: string, args: any..., callback: (err: Error, result: any) -> void)`
+There’re multiple types of hooks which run tapped functions a bit differently. They are described [in the Tapable docs](https://github.com/webpack/tapable#hook-types).
```

Comment:
That anchor does not longer exists. https://github.com/webpack/tapable should be fine

Reply:
This depends on https://github.com/webpack/tapable/pull/63 – the anchor will go live as soon as that PR is merged.

If it’s unlikely the PR would be merged soon, just tell – I’ll remove the anchor :–)